addons/mass_mailing/models/mailing.py

```diff
@@ -234,6 +235,15 @@ def _get_default_mail_server_id(self):
         'The A/B Testing Percentage needs to be between 0 and 100%'
     )]
 
+    @api.constrains('mailing_model_real', 'mailing_domain')
+    def _check_mailing_domain(self):
+        try:
+            for mailing in self:
+                domain = ast.literal_eval(mailing.mailing_domain)
+                self.env[mailing.mailing_model_real].search_count(domain)
```

Comment:
No need to actually do the query on postgres, simply crafting the `Query` object should be enough to sanity check the domain
```suggestion
                self.env[mailing.mailing_model_real]._search(domain)
```